src/EnergyPlus/WeatherManager.hh

```diff
@@ -67,205 +67,107 @@ namespace EnergyPlus {
-    extern int const EPlusWeatherFile;
-
-    extern int const ASHRAE_ClearSky;     // Design Day solar model ASHRAE ClearSky (default)
-    extern int const Zhang_Huang;         // Design Day solar model Zhang Huang
-    extern int const SolarModel_Schedule; // Design Day solar model (beam and diffuse) from user entered schedule
-    extern int const ASHRAE_Tau;          // Design Day solar model ASHRAE tau (per 2009 HOF)
-    extern int const ASHRAE_Tau2017;      // Design Day solar model ASHRAE tau (per 2013 and 2017 HOF)
-
-    extern int const DDHumIndType_WetBulb;   // Design Day Humidity Indicating Type = Wetbulb (default)
-    extern int const DDHumIndType_DewPoint;  // Design Day Humidity Indicating Type = Dewpoint
-    extern int const DDHumIndType_Enthalpy;  // Design Day Humidity Indicating Type = Enthalpy
-    extern int const DDHumIndType_HumRatio;  // Design Day Humidity Indicating Type = Humidity Ratio
-    extern int const DDHumIndType_RelHumSch; // Design Day Humidity Indicating Type = relhum schedule
-    extern int const DDHumIndType_WBProfDef; // Design Day Humidity Indicating Type = Wetbulb default profile
-    extern int const DDHumIndType_WBProfDif; // Design Day Humidity Indicating Type = Wetbulb difference profile
-    extern int const DDHumIndType_WBProfMul; // Design Day Humidity Indicating Type = Wetbulb multiplier profile
-    extern int const DDHumIndType_Count;     // # of DDHumIndTypes
-
-    extern int const DDDBRangeType_Default;    // Design Day DryBulb Range Type = Default Multipliers
-    extern int const DDDBRangeType_Multiplier; // Design Day DryBulb Range Type = Multiplier Schedule
-    extern int const DDDBRangeType_Difference; // Design Day DryBulb Range Type = Difference Schedule
-    extern int const DDDBRangeType_Profile;    // Design Day DryBulb Range Type = Temperature Profile
-
-    extern int const WP_ClarkAllenModel;  // Use Clark and Allen model for sky emissivity calculation
-    extern int const WP_ScheduleValue;  // User entered Schedule value for Weather Property
-    extern int const WP_DryBulbDelta;   // User entered DryBulb difference Schedule value for Weather Property
-    extern int const WP_DewPointDelta;  // User entered Dewpoint difference Schedule value for Weather Property
-    extern int const WP_BruntModel;  // Use Brunt model for sky emissivity calculation
-    extern int const WP_IdsoModel;   // Use Isdo model for sky emissivity calculation
-    extern int const WP_BerdahlMartinModel;  // Use Berdahl & Martin model for sky emissivity calculation
-    extern int const WP_SkyTAlgorithmA; // place holder
-
-    extern int const GregorianToJulian; // JGDate argument for Gregorian to Julian Date conversion
-    extern int const JulianToGregorian; // JGDate argument for Julian to Gregorian Date conversion
+    enum class DateType
+    {
+        InvalidDate = -1,
+        MonthDay = 1,
+        NthDayInMonth = 2,
+        LastDayInMonth = 3
+    };
```

Comment:
Turned these int variables all into `enum`s. It makes the code it affects definitely more readable.